### ISSUE TITLE
Regression: "TODO buf_write_value_bytes maybe type" since f247a9054 (4th March)

### ISSUE THREAD

**Issue by metaleap** [ziglang/zig#4674]
The nightly from https://github.com/ziglang/zig/commit/6cbd1ac51af4300ef11373d16771cf011fb6e572 still compiles `github.com/meta-leap/langserv` repo, but starting with any nightly from https://github.com/ziglang/zig/commit/f247a905419f47f794837c3818636553eebb929c or newer `zig build` prints:

```
TODO buf_write_value_bytes maybe type
Unable to dump stack trace: debug info stripped
dummylangserver...The following command terminated unexpectedly:
/home/_/a/zig/zig build-exe /home/_/c/z/langserv/dummylangserver/main.zig --cache-dir /home/_/c/z/langserv/zig-cache --name dummylangserver --main-pkg-path /home/_/c/z --cache on 

Build failed. The following command failed:
/home/_/c/z/langserv/zig-cache/o/xvVeGLArhljJxsswi2Np-jE0aFqC6kTF0kfj9G_N-kEzLzRaN7x5k9oneUVD77RT/build /home/_/a/zig/zig /home/_/c/z/langserv /home/_/c/z/langserv/zig-cache
```

Further drilling down into the failed commands only yields another

```
TODO buf_write_value_bytes maybe type
Unable to dump stack trace: debug info stripped
Aborted
```

(Dependencies of `meta-leap/langserv` are `meta-leap/jsonic` + `meta-leap/zigsess` + `meta-leap/zag` --- impossible to figure out the exact minimal subset of that sizable code-base that would still repro this regression, I tried for a bit to figure out a way.)

**Comment by metaleap** [comment]
I was too impatient and didn't want to stay on 6cbd1ac for too long. So I figured out surgically what to remove from `jsonic` and `langserv` repos in order to built on current latest nightly.

This means these repos can no longer be used to provoke the issue: **so feel free to close this** or leave open to keep a reminder for anyone to some day look into the code-paths that lead down to a `TODO buf_write_value_bytes maybe type` situation.

As a hint, the removals that matter where basically the contenders that also triggered #4380 --- back then I had to make those `fn`-typed fields nullable to proceed (with no real need to do so), now I just ditched them and hardcoded my default behaviors, and presto. Above issue was gone. The commit https://github.com/meta-leap/jsonic/commit/31aea7d2e95131d944b02b9f6054eb972ae75b9c illustrates the changes that put me back on the bleeding edge build of zig.
